src/base-apis.js

```diff
@@ -1071,6 +1071,10 @@ MatrixBaseApis.prototype.searchUserDirectory = function(opts) {
  * @param {string=} opts.name   Name to give the file on the server. Defaults
  *   to <tt>file.name</tt>.
  *
+ * @param {boolean=} opts.omitFilename if true will not send the filename,
+ *   e.g for encrypted file uploads where filename leaks are undesirable.
+ *   Defaults to false.
```

Comment:
I'm going to bang my, "positive arguments" drum again, ie. `if (includeFilename)` vs `if (!omitFilename)`

Reply:
But then it's a breaking change unless you mean to default it to true

Reply:
yep, suggesting defaulting to true

Reply:
changed that for you :)